### PR TITLE
Send GPS logging status as part of SYS_STATUS

### PR DESCRIPTION
**Pull request by WickedShell**
This allows a GPS which is responsible for logging to flag to a GCS when logging has failed. We will only send this down if the `GPS_RAW_DATA` parameter is not 0, so the user must have opted into it. This allows you to assess when a GPS has lost access to it's SD card for some reason.

This shows up as the logging bit to the GCS, which I think is consistent with the way we already use logging bits. IE a bad accelerometer health warning just tells us any one of our accelerometers has a problem, it doesn't refer to the primary one.

### CODE REVIEW

File: libraries/AP_GPS/AP_GPS.h [hunk 1]
```diff
@@ -431,6 +431,11 @@ class AP_GPS
     // returns true if all GPS instances have passed all final arming checks/state changes
     bool prepare_for_arming(void);
 
+    // returns true if all GPS drivers are prefroming their logging appropriately
```
Comment (peterbarker) [inline libraries/AP_GPS/AP_GPS.h]: Sorry, two twiddles doesn't make this correct.

File: libraries/GCS_MAVLink/GCS.cpp [hunk 2]
```diff
@@ -155,13 +156,14 @@ void GCS::update_sensor_status_flags()
...
+    const AP_GPS &gps = AP::gps();
+    if (logger.logging_present() || gps.drivers_should_log()) {  // primary logging only (usually File)
         control_sensors_present |= MAV_SYS_STATUS_LOGGING;
     }
-    if (logger.logging_enabled()) {
+    if (logger.logging_enabled() || gps.drivers_should_log()) {
         control_sensors_enabled |= MAV_SYS_STATUS_LOGGING;
     }
-    if (!logger.logging_failed()) {
+    if (!logger.logging_failed() && gps.logging_healthy()) {
```
Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS.cpp]: The methods were named quite deliberately for their intended purpose.  Logging isn't considered failed if we are not intending to log.  Asserting logging is healthy if you're not intending to right isn't possible, thus the "negative" aspect of the function.

Comment (WickedShell) [reply libraries/GCS_MAVLink/GCS.cpp]: Look at what the logging_healthy() function does, if you aren't intending to log then it's considered healthy.

**Comment by peterbarker** [comment]
On Thu, 14 Nov 2019, WickedShell wrote:

> Look at what the logging_healthy() function does, if you aren't intending to log then it's considered healthy.

Yes, and that's bad - as you really don't know if the logging system can 
actually do its job.

**Pull request merged**